drivers/interrupt_controller/intc_gic.c

```diff
@@ -11,19 +11,80 @@
  *       GICv3 interface is not supported.
  */
 
-#include <device.h>
 #include <sw_isr_table.h>
-#include <irq_nextlevel.h>
 #include <dt-bindings/interrupt-controller/arm-gic.h>
 #include <drivers/interrupt_controller/gic.h>
 
 #if CONFIG_GIC_VER >= 3
 #error "GICv3 and above are not supported"
 #endif
 
-struct gic_ictl_config {
-	u32_t isr_table_offset;
-};
+void arm_gic_irq_enable(unsigned int irq)
+{
+	int int_grp, int_off;
+
+	int_grp = irq / 32;
+	int_off = irq % 32;
+
+	sys_write32((1 << int_off), (GICD_ISENABLERn + int_grp * 4));
+}
+
+void arm_gic_irq_disable(unsigned int irq)
+{
+	int int_grp, int_off;
+
+	int_grp = irq / 32;
+	int_off = irq % 32;
+
+	sys_write32((1 << int_off), (GICD_ICENABLERn + int_grp * 4));
+}
+
+int arm_gic_irq_is_enabled(unsigned int irq)
```

Comment:
Can we add here something like:
```
+       /* SGIs are never available */
+       if (irq < 16)
+               return 0;
+
```
The problem comes from the `tests/kernel/interrupt/arch.interrupt` test that is failing the assert at https://github.com/zephyrproject-rtos/zephyr/blob/master/arch/common/sw_isr_common.c#L19 when `irq == 0`.

Reply:
At least, for PL-390 and GIC-400, the SGIs (0-15) are always enabled and this is an expected behaviour (`arm_gic_irq_is_enabled` should report the actual enable status of the IRQ, which is `true`).

The real problem is that the dynamic interrupt test scans the SW ISR table for the first spurious ISR entry and this happens to be the IRQ 0, which is an SGI that cannot be disabled on the GIC-400.

IMO, this problem should be addressed separately in the issue #22670. After all, without proper GIC awareness in the interrupt test, removing the `interrupt` test ignore tag would only provide a false sense of security.